Repo: v8 failing to publish: Cannot find module 'tools/release/changelog-renderer'

### Suggestion

Example from https://github.com/typescript-eslint/typescript-eslint/actions/runs/9475827919/job/26108105626?pr=9165 -> https://cloud.nx.app/runs/7TN5XXmVPt:

```plaintext
> nx release version 8.0.0-alpha.31


 NX   There was an error when resolving the configured changelog renderer at path: tools/release/changelog-renderer


 NX   Cannot find module 'tools/release/changelog-renderer'

Require stack:
- /home/runner/work/typescript-eslint/typescript-eslint/node_modules/nx/src/command-line/release/utils/resolve-changelog-renderer.js
- /home/runner/work/typescript-eslint/typescript-eslint/node_modules/nx/src/command-line/release/config/config.js
- /home/runner/work/typescript-eslint/typescript-eslint/node_modules/nx/src/command-line/release/version.js
- /home/runner/work/typescript-eslint/typescript-eslint/node_modules/nx/src/command-line/release/command-object.js
- /home/runner/work/typescript-eslint/typescript-eslint/node_modules/nx/src/command-line/nx-commands.js
- /home/runner/work/typescript-eslint/typescript-eslint/node_modules/nx/bin/init-local.js
- /home/runner/work/typescript-eslint/typescript-eslint/node_modules/nx/bin/nx.js
Pass --verbose to see the stacktrace.

file:///home/runner/work/typescript-eslint/typescript-eslint/packages/repo-tools/node_modules/execa/lib/error.js:60
    error = new Error(message);
            ^

Error: Command failed with exit code 1: npx nx release version 8.0.0-alpha.31
```

cc @JamesHenry

--- a/.yarn/patches/nx-npm-19.3.0-canary.20240611-1600875-bd96520563.patch
+++ b/.yarn/patches/nx-npm-19.3.0-canary.20240611-1600875-bd96520563.patch
@@ -1,8 +1,8 @@
 diff --git a/src/command-line/release/utils/resolve-changelog-renderer.js b/src/command-line/release/utils/resolve-changelog-renderer.js
-index 8c6fa468633703fcdbe491c08e509d10574cd75d..b1a669d22097dde0c88c7c8d5a956159b246b33c 100644
+index 8c6fa468633703fcdbe491c08e509d10574cd75d..c36c55b75bffb22a173ec8c5ad5a37abd76b24e4 100644
 --- a/src/command-line/release/utils/resolve-changelog-renderer.js
 +++ b/src/command-line/release/utils/resolve-changelog-renderer.js
-@@ -13,10 +13,10 @@ function resolveChangelogRenderer(changelogRendererPath) {
+@@ -13,11 +13,11 @@ function resolveChangelogRenderer(changelogRendererPath) {
      let changelogRenderer;
      let cleanupTranspiler = () => { };
      try {
@@ -10,10 +10,12 @@ index 8c6fa468633703fcdbe491c08e509d10574cd75d..b1a669d22097dde0c88c7c8d5a956159
 -        if (rootTsconfigPath) {
 -            cleanupTranspiler = (0, register_1.registerTsProject)(rootTsconfigPath);
 -        }
+-        const r = require(interpolatedChangelogRendererPath);
 +        // const rootTsconfigPath = (0, typescript_1.getRootTsConfigPath)();
 +        // if (rootTsconfigPath) {
 +        //     cleanupTranspiler = (0, register_1.registerTsProject)(rootTsconfigPath);
 +        // }
-         const r = require(interpolatedChangelogRendererPath);
++        const r = require(require('node:path').join(interpolatedChangelogRendererPath));
          changelogRenderer = r.default || r;
      }
+     catch (err) {

--- a/nx.json
+++ b/nx.json
@@ -6,10 +6,10 @@
     "changelog": {
       "workspaceChangelog": {
         "createRelease": "github",
-        "renderer": "tools/release/changelog-renderer"
+        "renderer": "{workspaceRoot}/tools/release/changelog-renderer"
       },
       "projectChangelogs": {
-        "renderer": "tools/release/changelog-renderer"
+        "renderer": "{workspaceRoot}/tools/release/changelog-renderer"
       }
     },
     "version": {

--- a/package.json
+++ b/package.json
@@ -139,7 +139,6 @@
     "tmp": "0.2.1",
     "tsx": "^4.7.2",
     "typescript": "5.4.5",
-    "nx@18.3.5": "patch:nx@npm%3A18.3.5#./.yarn/patches/nx-npm-18.3.5-58c0bedf91.patch",
     "eslint-plugin-eslint-plugin@^5.5.0": "patch:eslint-plugin-eslint-plugin@npm%3A5.5.1#./.yarn/patches/eslint-plugin-eslint-plugin-npm-5.5.1-4206c2506d.patch",
     "@nx/eslint@19.3.0-canary.20240611-1600875": "patch:@nx/eslint@npm%3A19.3.0-canary.20240611-1600875#./.yarn/patches/@nx-eslint-npm-19.3.0-canary.20240611-1600875-f5cb695794.patch",
     "nx@19.3.0-canary.20240611-1600875": "patch:nx@npm%3A19.3.0-canary.20240611-1600875#./.yarn/patches/nx-npm-19.3.0-canary.20240611-1600875-bd96520563.patch"

